### ISSUE TITLE
[FEATURE] Improvements for including Embedded Audio Player in the component header

### ISSUE THREAD

**Issue by cjolson** [carlonicora/obsidian-rpg-manager#230]
## Scope

I'm messing with music and have figured out a set up I like to include an embedded player in the Header for relevant components. This thread is two things in one:
1. [Feature Request] Some improvements to the default `model` lists (more details below) and allow selection of Music in the Relationship Manager.
2. [Question] Do you anticipate any future conflicts in my setup? I suspect I'm not doing things as expected/intended, so I don't want to break things later if I proceed to add all my audio files the way described below.

The need my set up is solving for:
> When running a `Scene` or traveling to a `Location`, I want easy to easily control music that is relevant to the situation at hand. I have various mp3s (most from [Tabletop Audio](https://tabletopaudio.com)) stored in my Vault. Ideally, I could access a media player embedded in the relevant notes for these files at the bottom of the `header model` and right before my note contents.

#### TL;DR (summary of the details below):
- Move the list for `Music` components to the head `model` (as opposed to the foot) for `Scenes` so that you can click "Play" when you start the scene.
- Add `Music` to the default list for `Location` (and any other components it could relevant?) so that you can access music that is location-dependent, but not scene-dependent. (This is particularly helpful if your party goes to the "Yawning Portal" when you did not expect it and don't have a scene prepped).
  - _Bonus:_ Add it as the last list in the `model` so it is positioned immediately before your notes for the Location. This will make it easier to access later if you have scrolled down and need to scroll back up to manage audio controls. 
- Allow the ability to select `Music` in the Relationship Manager so it is easy to add.

### Describe the solution you'd like

For my set up, I am using the [Media Extended plugin](https://github.com/aidenlx/media-extended) for the feature that let's you play the audio on loop from an embedded player. ~~You can include a non-loopable embed audio player with Obisidian's default player by adding `![[filename.mp3]]` in a note.~~ Correction, apparently the default embedded player does _not_ display in the codeblock. 

In the Data code block, I include the embedded link to the mp3 file in my Vault as the `url`. I initially included the description in the `synopsis`, but the media player is too wide to have both displayed in the Relationship List section of related notes, so I appended the description to the `url` instead and turned off `synopsis` for `Music` Lists in options. 

Example:
````
```RpgManagerData
data: 
  synopsis: ""
  complete: false
  url: "![[23_The_Slaughtered_Ox.mp3#loop]] A horn of mead, a bowl of stew and a hook to hang your 50 feet of rope, The Slaughtered Ox has been catering to the adventuring class for eons. (Ambience + Music)"
```
````

What it looks like in the `Music` note:

<img width="400" alt="Screenshot 2022-11-05 at 12 15 33 PM" src="https://user-images.githubusercontent.com/18406537/200129519-96392896-139b-474c-8649-0f406aa77128.png">

This is where an update could be super helpful:
- To include the Music to a `Location`, I have to add `musics: {}` to the `model` lists manually and include a link to the `Music` component file in the body of the note. This is where my main feature requests come in:
  -  _**I would like to (1) add `Music` component relationships to locations from the Relationship Manager and (2) for `musics: {}` to be include in the `model` lists by default (last in the order of all the lists so it's most accessible by scrolling up later if I need to access the controls)**_.
- To add Music to a `Scene`, I also can't seem to add it from the Relationship Manager, so I have to include it by linking to it in the body of the note. However, I also notice that the `model` lists for this component populates at the foot of the note in a separate `model` (after my notes and customs template) instead of the head. For other `model` lists, this has been fine – **_but for music controls, I think it would be ideal to include it in the `model` at the top of the note so I can press play when I begin the scene_** (if other lists are included in the first `model` I would still include music in the last of those lists for the reasons included above).

Example of how I accomplish this manually in a `Scene`:
````
```RpgManager
models: 
  header: true
  lists: 
    musics: 
      relationship: "unidirectional"
```
````

And what it looks like for a `Scene` (sidenote: that this is the shortest width I can get from the Media Extended plugin's audio player):

<img width="400" alt="Screenshot 2022-11-05 at 12 42 44 PM" src="https://user-images.githubusercontent.com/18406537/200131428-ec1455e2-478d-40e5-8b54-5578f29058d6.png">

### Alternatives

Right now, my manual setup works (unless you think I'm gonna break the updater). If you have ideas for better alternatives though, go for it.

**Comment by cjolson** [comment]
After testing it some more, there actually seems to be a bit of a hiccup in playing the music from the `codeblock`. It will generally play uninterrupted just fine – but once you start typing in any notes that are apart of RPG Manager the music will stop and _sometimes_ get stuck on a buffer when you try to play again (but not always). I do see the media players kinda 'reload' when they stop playing, so I wonder if it's RPG Manager live-updating as I when it senses edits?

I don't see any errors thrown in developer tools. 

#### Workaround:
- I don't run into this problem if I open the `Music` note and play it from there (from the `link` section of the component or a player embedded in the body; both work fine). 
- I can also [embed a block](https://help.obsidian.md/How+to/Link+to+blocks) from the body of the `Music` note in my `Scene`/etc. notes and it will work as well.

Not sure if making it play from the `codeblock` is feasible. I think it would be awesome if it could be supported. Though it might be more effort than it initially seemed.

**Comment by carlonicora** [comment]
@cjolson the issue with playing the music from the note itself is that many things may refresh the note, hence restarting the music.
An option would be to create a separate view in the right leaf, containing a player, and using the note to trigger the start/stop of the music.

Something for me to think about (_and to find a flexible enough media player to embed in obsidian_)

**Comment by cjolson** [comment]
@carlonicora Sounds good. For now I've taken to embedding a block in the scene and playing it from the embed. If I notice other hiccups or issues, I'll keep you updated.

<img width="500" alt="Screenshot 2022-11-05 at 7 07 51 PM" src="https://user-images.githubusercontent.com/18406537/200145097-d6d86ddb-3cb9-426f-a462-5d4133b737f6.png">

**Comment by sigrunixia** [comment]
Rolling this into internal rewriting list.
